### DISCUSSION TITLE
Interaction between icudtl.dat with skshaper on windows?

### DISCUSSION THREAD

**Discussion by HinTak** [rust-skia/rust-skia#1041] [General]
I am coming over from skia-python and wonder if you guy experiences something similar?

https://issues.skia.org/371224771

Basically having an icudtl.dat breaks SkShaper, although it is a must for SkParagraph. It seems that SkShaper was happy without icudtl.dat, but adding it in the load path actually break skshaper on windows.

Or maybe I am not getting a correct icudtl.dat - I get it upstream, but maybe there are google-specific changes that I must get it from the Google-forked icu repo? What are you guys experience on using skshaper on windows?

**Comment by pragmatrix** [comment]
Hello @HinTak, 

we had related problems (#168, #557) , and therefore decided to embed `icudtl.dat` on Windows in #486, like it's done on Linux and macOS. This change needed additional patches to Skia. But since then I can't remember that we had any more problems with a missing `icudtl.dat`.

**Comment by HinTak** [comment]
Hi @pragmatrix thanks for responding. I found the issue - upstream basically periodically bump the icu version up: 

https://github.com/kyamagu/skia-python/pull/270/commits/2b6ab54ea4c109583f0b28c67dfb246d00a0d01c

It last went from 69 to 74 between m125 and m126 . I tried using current (75) which is probably too new. This is a bit painful as we don't really want to patch skia too much, nor keep a large build artefact like this.
